packages/nodes-base/nodes/Notion/shared/descriptions/DatabasePageDescription.ts

```diff
@@ -601,15 +601,15 @@ export const databasePageFields: INodeProperties[] = [
 						type: 'regex',
 						properties: {
 							regex:
-								'(?:https|http)://www.notion.so/(?:[a-z0-9-]{2,}/)?(?:[a-zA-Z0-9-]{2,}-)?([0-9a-f]{8}[0-9a-f]{4}4[0-9a-f]{3}[89ab][0-9a-f]{3}[0-9a-f]{12}).*',
+								'(?:https|http)://www.notion.so/(?:[a-z0-9-]{2,}/)?(?:[a-zA-Z0-9-]{1,}-)?([0-9a-f]{8}[0-9a-f]{4}4[0-9a-f]{3}[89ab][0-9a-f]{3}[0-9a-f]{12}).*',
```

Comment:
This regex expects the page ID's thirteenth character to always be a `4` (i.e., `...[0-9a-f]{4}4[0-9a-f]{3}...`), which is an incorrect assumption. For example, `https://www.notion.so/tenant-id/fffb95d3060b80309027eb9c99605ec3` and `https://www.notion.so/fffb95d3060b80309027eb9c99605ec3` are valid Notion URL's. As such, this (and the other, similar regex's in this PR) should be fixed before approving/merging.

Additionally, I'd recommend loosening up all the Notion regex's in this PR. While validing user input strictly is a noble endeavor, these regex's make assumptions about Notion's internal infra, which is undocumented and subject to change without notice. Worse, when these assumptions prove wrong at runtime, users like [artildo](https://community.n8n.io/u/artildo) and I are left with a broken n8n Notion integration for weeks without a workaround[^1]. Holistically, the benefits of validating user input so strictly are negated by a broken Notion integration, so the regex's need to be loosened up.

[^1]: Short of making raw HTTP requests, which defeats the purpose of n8n.

Reply:
@jerzygangi the regex I believe was originally set up to make sure the ID also matches a UUIDv4 which is what Notion uses for a Page, You are not wrong through for a Database Page Notion just uses a UUID and they don't document the version like they do for the Page.

I will get these updated for the fields that do not specifically mention the v4 ID.

Page Properties: https://developers.notion.com/reference/page#page-object-properties
Database Page Properties: https://developers.notion.com/reference/database

I suspect we originally spotted the v4 type and ran with it.